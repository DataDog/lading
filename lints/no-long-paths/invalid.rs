// Cases that SHOULD trigger the lint (3+ segments, not in use statements). If
// custom_lints does NOT trigger on this file the tool is NOT working as
// intended.

fn example_violations() {
    // 3 segments - should trigger
    let config = target::Config::Pid;
    let server = generator::Server::new();
    let metrics = json::MetricKind::Counter;

    // 4 segments - should trigger
    let cache = block::Cache::fixed_with_max_overhead();
    let sampler = cgroup::Sampler::new();

    // 5+ segments - should trigger
    let very_long = foo::bar::baz::qux::deeply::nested::call();
    let another = project::module::submodule::function::call::chain();

    // Mixed with valid code
    let file = fs::File::open("valid"); // 3 segments - should trigger
    let violation = target::Config::Docker; // 3 segments - should trigger
}

// More examples in different contexts
impl SomeStruct {
    fn method(&self) {
        some::long::path::here().field; // should trigger
    }
}

// In match statements
fn match_example(value: SomeEnum) {
    match value {
        some::variant::Path => {} // should trigger
        other => {}
    }
}

// Complex expressions that should trigger
fn complex_cases() {
    // Method chaining - the path segments should trigger
    let result = some::long::path::to::function()
        .map(|x| x.process())
        .unwrap_or_default();

    // Nested calls
    let value = outer::function(inner::long::path::call());

    // Array/struct initialization
    let config = Config {
        field: some::module::Type::default(),
        other: valid_two::segments(),
    };

    // Turbofish syntax
    let parsed = some::parser::module::<String>::parse();
}

// Macros with long paths
fn macro_violations() {
    some_macro!(with::long::path::args);
    format!("string with {}", some::path::variable);
}
