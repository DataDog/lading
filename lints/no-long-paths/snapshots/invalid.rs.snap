error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
  ┌─ lints/no-long-paths/invalid.rs:7:18
  │
7 │     let config = target::Config::Pid;
  │                  ^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
  ┌─ lints/no-long-paths/invalid.rs:8:18
  │
8 │     let server = generator::Server::new();
  │                  ^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
  ┌─ lints/no-long-paths/invalid.rs:9:19
  │
9 │     let metrics = json::MetricKind::Counter;
  │                   ^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:12:17
   │
12 │     let cache = block::Cache::fixed_with_max_overhead();
   │                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:13:19
   │
13 │     let sampler = cgroup::Sampler::new();
   │                   ^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:16:21
   │
16 │     let very_long = foo::bar::baz::qux::deeply::nested::call();
   │                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:16:21
   │
16 │     let very_long = foo::bar::baz::qux::deeply::nested::call();
   │                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:16:21
   │
16 │     let very_long = foo::bar::baz::qux::deeply::nested::call();
   │                     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:16:21
   │
16 │     let very_long = foo::bar::baz::qux::deeply::nested::call();
   │                     ^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:16:21
   │
16 │     let very_long = foo::bar::baz::qux::deeply::nested::call();
   │                     ^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:17:19
   │
17 │     let another = project::module::submodule::function::call::chain();
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:17:19
   │
17 │     let another = project::module::submodule::function::call::chain();
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:17:19
   │
17 │     let another = project::module::submodule::function::call::chain();
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:17:19
   │
17 │     let another = project::module::submodule::function::call::chain();
   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:20:16
   │
20 │     let file = fs::File::open("valid"); // 3 segments - should trigger
   │                ^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:21:21
   │
21 │     let violation = target::Config::Docker; // 3 segments - should trigger
   │                     ^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:27:9
   │
27 │         some::long::path::here().field; // should trigger
   │         ^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:27:9
   │
27 │         some::long::path::here().field; // should trigger
   │         ^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:34:9
   │
34 │         some::variant::Path => {} // should trigger
   │         ^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:42:18
   │
42 │     let result = some::long::path::to::function()
   │                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:42:18
   │
42 │     let result = some::long::path::to::function()
   │                  ^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:42:18
   │
42 │     let result = some::long::path::to::function()
   │                  ^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:47:33
   │
47 │     let value = outer::function(inner::long::path::call());
   │                                 ^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:47:33
   │
47 │     let value = outer::function(inner::long::path::call());
   │                                 ^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:51:16
   │
51 │         field: some::module::Type::default(),
   │                ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:51:16
   │
51 │         field: some::module::Type::default(),
   │                ^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/invalid.rs:56:18
   │
56 │     let parsed = some::parser::module::<String>::parse();
   │                  ^^^^^^^^^^^^^^^^^^^^

Error: 27 error(s) found in code.
Help: Scan succeeded and found error level diagnostics in the codebase.
